cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -521,9 +521,6 @@ class TestS3FS : public S3TestMixin {
   void TestOpenOutputStream() {
     std::shared_ptr<io::OutputStream> stream;
 
-    // Nonexistent
-    ASSERT_RAISES(IOError, fs_->OpenOutputStream("nonexistent-bucket/somefile"));
```

Comment:
Why this is being removed?

Reply:
See the PR description ("User facing changes") and the discussion below. With this optimization we may throw later if the bucket does not exist because the `Open` of the output stream does not trigger an S3 request any more.